guides/source/testing.md

```diff
@@ -859,292 +846,440 @@ users(:david, :steve)
+associated logic. You can test this logic using the assertions and fixtures that
+we've explored in the sections above.
 
-Rails model tests are stored under the `test/models` directory. Rails provides
-a generator to create a model test skeleton for you.
+Rails model tests are stored under the `test/models` directory. Rails provides a
+generator to create a model test skeleton for you.
 
 ```bash
 $ bin/rails generate test_unit:model article title:string body:text
 create  test/models/article_test.rb
-create  test/fixtures/articles.yml
 ```
 
-Model tests don't have their own superclass like `ActionMailer::TestCase`. Instead, they inherit from [`ActiveSupport::TestCase`](https://api.rubyonrails.org/classes/ActiveSupport/TestCase.html).
+Model tests don't have their own superclass like `ActionMailer::TestCase`.
+Instead, they inherit from
+[`ActiveSupport::TestCase`](https://api.rubyonrails.org/classes/ActiveSupport/TestCase.html).
 
-System Testing
---------------
+Functional Testing for Controllers
+----------------------------------
 
-System tests allow you to test user interactions with your application, running tests
-in either a real or a headless browser. System tests use Capybara under the hood.
+In Rails, a focus on testing the various actions of a controller is a form of
+writing functional tests. Remember, your controllers handle the incoming web
+requests to your application and eventually respond with a rendered view. 
 
-For creating Rails system tests, you use the `test/system` directory in your
-application. Rails provides a generator to create a system test skeleton for you.
+When writing functional tests, you are focusing on testing how your actions
+handle the requests and the expected result or response. Functional controller
+tests are sometimes used in cases where system tests are not appropriate, e.g.,
+to confirm an API response.
 
-```bash
-$ bin/rails generate system_test users
-      invoke test_unit
-      create test/system/users_test.rb
-```
+### What to Include in Your Functional Tests
 
-Here's what a freshly generated system test looks like:
+You could test for things such as:
 
-```ruby
-require "application_system_test_case"
+* was the web request successful?
+* was the user redirected to the right page?
+* was the user successfully authenticated?
+* was the correct information displayed in the response?
 
-class UsersTest < ApplicationSystemTestCase
-  # test "visiting the index" do
-  #   visit users_url
-  #
-  #   assert_selector "h1", text: "Users"
-  # end
-end
-```
+The easiest way to see functional tests in action is to generate a controller
+using the scaffold generator:
 
-By default, system tests are run with the Selenium driver, using the Chrome
-browser, and a screen size of 1400x1400. The next section explains how to
-change the default settings.
+```bash
+$ bin/rails generate scaffold_controller article title:string body:text
```

Comment:
Thanks, good spot - actually I think maybe just the `title:string` and `body:text` is wrong, as we've generated the model in the last section, so it would make sense to be generating the controller part here! I think it's a bit of  bad copy paste somewhere along the line!